src/data-model-providers/codegen/ServerClusterInterfaceRegistry.cpp

```diff
@@ -55,9 +56,9 @@ CHIP_ERROR ServerClusterInterfaceRegistry::Register(ServerClusterRegistration &
     // items. We preserve this however we may want to make this optional at some point in time.
     VerifyOrReturnError(Get(path) == nullptr, CHIP_ERROR_DUPLICATE_KEY_ID);
 
-    if (mContextIsValid)
+    if (mContext.has_value())
     {
-        ReturnErrorOnFailure(entry.serverClusterInterface->Startup(&mContext));
+        ReturnErrorOnFailure(entry.serverClusterInterface->Startup(&*mContext));
```

Comment:
Followup: Startup should also be taking a context reference, since null is not allowed there, right?

Reply:
Unsure about this as reference may encourage people to "create a copy" and that uses more RAM. The intent was to say "you may store this pointer and it is usable for the duration of the cluster validity period".

Do you think reference is ok as long as DefaultServerCluster just takes a pointer to the reference?

Reply:
This is precisely why C++ memory lifetimes suck so bad 😅

In general from what I've seen pointer/reference only differentiate nullable vs not; any assumptions made about lifetimes from it being a pointer vs reference are almost always wrong 🤣

Does `shared_ptr` or `unique_ptr` make sense here or is that disallowed?

Reply:
I strongly prefer pointers as they do avoid copies, and in any case, if someone passes nullptr, there's bigger issues at play.

Reply:
I don't understand why someone would want to "create a copy" to pass a reference, any more so than passing a pointer....  And yes, C++ lifetimes are terrible, and yes shared_ptr/unique_ptr, since they require heap allocation, are not OK here.

Reply:
Will create a followup change for this, we can review in the PR if it makes sense or not. The change is expected to be somewhat small anyway.